packages/@aws-cdk/aws-cloudwatch/lib/insight-rule.ts

```diff
@@ -0,0 +1,366 @@
+  }
+
+  /**
+   * Convenience function to generate an EQUAL_TO Filter
+   */
+  public static equalTo(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.EQUAL_TO, target);
+  }
+
+  /**
+   * Convenience function to generate a NOT_EQUAL_TO Filter
+   */
+  public static notEqualTo(match: string, target: number): Filter {
+    return new Filter(match, FilterOperator.NOT_EQUAL_TO, target);
+  }
+
+  /**
+   * Convenience function to generate an IS_PRESENT Filter
+   */
+  public static isPresent(match: string, target: boolean): Filter {
+    return new Filter(match, FilterOperator.IS_PRESENT, target);
+  }
+
+  /**
+   * The log event field to assess in the filter
+   */
+  readonly match: string;
+
+  /**
+   * The type of Filter to apply
+   */
+  readonly operator: FilterOperator;
+
+  /**
+   * The target field to compare the `match` against
+   */
+  readonly target: string[] | number | boolean;
+
+  private constructor(match: string, operator: FilterOperator, target: string[] | number | boolean) {
+    this.match = match;
+    this.operator = operator;
+    this.target = target;
+  }
+}
+
+/**
+ * The applicable Log Formats for the Insight Rule
+ */
+export enum LogFormat {
+  /**
+   * JSON format
+   */
+  JSON = 'JSON',
+
+  /**
+   * CLF format
+   */
+  CLF = 'CLF',
+}
+
+/**
+ * The options to aggregate the log events to evaluate contribution for the Insight Rule
+ */
+export enum AggregateOptions {
+  /**
+   * Aggregate contribution by summing a given field
+   */
+  SUM = 'Sum',
+
+  /**
+   * Aggregate contribution by the count of a contributor's appearance in events
+   */
+  COUNT = 'Count',
+}
+
+/**
+ * The properties for an Insight Rule
+ */
+export interface InsightRuleProps {
+  /**
+   * Option to Aggregate the Log Events for the Insight Rule to assess contribution
+   *
+   * @default - COUNT
+   */
+  readonly aggregateOn?: AggregateOptions,
+
+  /**
+   * Whether or not the Insight Rule is enabled
+   *
+   * @default true
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * The Filters to apply to the Insight Rule
+   *
+   * @default []
+   */
+  readonly filters?: Filter[],
+
+  /**
+   * The keys of the log event to identify a Contributor
+   */
+  readonly keys: string[],
+
+  /**
+   * The Log Format of the Insight Rule
+   *
+   * @default - JSON
+   */
+  readonly logFormat?: LogFormat,
+
+  /**
+   * The Log Groups to include in the report
+   */
+  readonly logGroupNames: string[],
+
+  /**
+   * The name of the Insight Rule
+   */
+  readonly insightRuleName: string,
+
+  /**
+   * The name of the Schema. Should usually be 'CloudWatchLogRule'
+   *
+   * @default 'CloudWatchLogRule'
+   */
+  readonly schemaName?: string,
+
+  /**
+   * The value to sum up if the Rule is set to aggregate on SUM
+   *
+   * @default none
+   */
+  readonly sumValue?: string,
+
+  /**
+   * The version of the Schema. Should usually be 1
```

Comment:
What does "should usually be 1" mean? For whom should it usually be one, and under what circumstances would it not be? As it stands, this doesn't tell me much.

Reply:
The value of Schema for a rule that analyzes CloudWatch Logs data must always be 
`{"Name": "CloudWatchLogRule", "Version": 1}`
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContributorInsights-RuleSyntax.html
I made it customizable anyways in case that changes or if someone knows something the docs aren't letting on

Reply:
k. i still don't like the language though. and, it is instantly wrong when schema version 2 is released. 

what is the default? is it 1, or "latest version available"? My guess is latest available.